### ISSUE TITLE
ShowPartial breaks the navigation dots.

### ISSUE THREAD

**Issue by fmonper1** [FormidableLabs/nuka-carousel#720]
# Bugs and Questions

### Describe Your Environment

React 16.13
Nuka 4.77

### Describe the Problem

I have implemented a slider using NukaCarousel. I use the showPartial to make it more intuivite for users, but when the option is turned on, the last dot in the navigation dots never gets activated.

My custom navigation dots:

```
  const renderDotControls = ({
    currentSlide,
    goToSlide,
    nextSlide,
    previousSlide,
    slideCount,
  }) => {
    if (slideCount === 1) return <></>;
    return (
      <div>
        {[...Array(slideCount)].map((sc, i) => (
          <div
            style={{ listStyleType: 'none', display: 'inline-block' }}
            key={i}
          >
            <div
              className={`${classes.dot} ${
                currentSlide === i ? classes.activeDot : ''
              }`}
              style={{
                width: '10px',
                height: '4px',
                borderRadius: '2px',
                margin: '0 3px',
              }}
            >
              {' '}
            </div>
          </div>
        ))}
      </div>
    );
  };
```

Expected behaviour: https://i.imgur.com/CwTZMcK.gif

Actual behaviour: https://imgur.com/x7YdD6R

**Comment by sarmeyer** [comment]
Hi @fmonper1 - thanks for submitting this issue. I am trying to reproduce the problem, but I'm having trouble. When I test out your custom PagingDots, I see their styles update correctly when active. I'm thinking I'm missing something...can you provide me with a little more information? What are you setting for `slidesToShow` and `slidesToScroll`? And are you using the `renderBottomCenterControls` helper method to set these custom Dots?

**Comment by fmonper1** [comment]
Yes, when using the renderBottomCenterControls the slidecount returns a decimal instead of an Int for the current index of the sliders, thus the custom navigation dots not being set to active correctly.

```
const NukaCarousel = props => {
  const { showPartial, showPartialAmount, ...rest } = props;
  const classes = useStyles();
  const mainHeightContainer = useRef();

  const renderDotControls = ({
    currentSlide,
    goToSlide,
    nextSlide,
    previousSlide,
    slideCount,
  }) => {
    if (slideCount === 1) return <></>;
    return (
      <div>
        {[...Array(slideCount)].map((sc, i) => (
          <div
            style={{ listStyleType: 'none', display: 'inline-block' }}
            key={i}
          >
            <div
              className={clsx(
                classes.dot,
                Math.round(currentSlide) === i && classes.activeDot
              )}
              style={{
                width: '6px',
                height: '6px',
                borderRadius: '50%',
                margin: '0 3px',
              }}
            >
              {' '}
            </div>
          </div>
        ))}
      </div>
    );
  };

  return (
    <Grid
      container
      className={classes.slider}
      alignItems="center"
      justify="center"
      id="mainHeightContainer"
      ref={mainHeightContainer}
    >
      <Grid item container className={classes.nuka}>
        <Carousel
          autoplay={false}
          disableEdgeSwiping
          slidesToShow={
            showPartial ? (showPartialAmount ? showPartialAmount : 1.1) : 1
          }
          defaultControlsConfig={{
            nextButtonClassName: classes.rightChevron,
            prevButtonClassName: classes.leftChevron,
            pagingDotsContainerClassName: classes.containerDots,
          }}
          renderCenterLeftControls={({ previousSlide, currentSlide }) =>
            currentSlide !== 0 ? (
              <Hidden smDown>
                <IconButton
                  size="small"
                  onClick={previousSlide}
                  className={classes.controlBtns}
                >
                  <Icon
                    path={mdiChevronDoubleLeft}
                    size={1}
                    className={classes.svgIcon}
                  />
                </IconButton>
              </Hidden>
            ) : null
          }
          renderCenterRightControls={({
            slideCount,
            nextSlide,
            currentSlide,
          }) =>
            currentSlide !== props.children - 1 && slideCount > 1 ? (
              <Hidden smDown>
                <IconButton
                  size="small"
                  onClick={nextSlide}
                  className={classes.controlBtns}
                >
                  <Icon
                    path={mdiChevronDoubleRight}
                    size={1}
                    className={classes.svgIcon}
                  />
                </IconButton>
              </Hidden>
            ) : null
          }
          renderBottomCenterControls={props => renderDotControls(props)}
          className={classes.sliderContainer}
          {...rest}
        >
          {props.children}
        </Carousel>
      </Grid>
    </Grid>
  );
};
```

I could fix this in my projects by rounding the currentSlide so that it returnes and Int

`Math.round(currentSlide) === i && classes.activeDot`

**Comment by sarmeyer** [comment]
Fixed in #734 - I'll release EOD tomorrow 11/6/20
